### PR TITLE
Feat/html templating

### PR DESCRIPTION
**Pull request by Reyth3**

### CODE REVIEW

File: src/pluto_templating/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    let out_dir = format!("{}", std::env::var("OUT_DIR").expect("No source path set."));
+    let mut engine = ructe::Ructe::new(PathBuf::from(out_dir)).unwrap();
+    match engine.compile_templates(templates_dir) {
+        Err(_) => return Ok(()),
+        _ => {}
+    };
+    match engine.statics() {
+        Err(_) => return Ok(()),
+        Ok(x) => {
+            let mut handle = x;
```
Comment (pu0238) [inline src/pluto_templating/src/lib.rs:16]: Did we can't add a mut to Ok abow?

File: src/pluto/src/view.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+///
+#[macro_export]
+macro_rules! render_view {
+    (
+        $view:path
+        $(, $arg:expr)*
+    ) => {
+        let mut headers = HashMap::new();
+        headers.insert(String::from("Content-Type"), String::from("text/html"));
+        let mut buffer: Vec<u8> = vec![];
```
Comment (pu0238) [inline src/pluto/src/view.rs:14]:
```suggestion
        let mut buffer: Vec<u8> = Vec::new();
```

File: src/pluto/src/view.rs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+///
+///
+#[macro_export]
+macro_rules! render_view {
+    (
+        $view:path
+        $(, $arg:expr)*
+    ) => {
+        let mut headers = HashMap::new();
+        headers.insert(String::from("Content-Type"), String::from("text/html"));
```
Comment (pu0238) [inline src/pluto/src/view.rs:13]: lets try to use .to_string() insted of String::from(

File: examples/src/http_server_html_template/views/index.rs.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
+
+@()
```
Comment (pu0238) [inline examples/src/http_server_html_template/views/index.rs.html:2]: do we need it to be empty?

File: examples/src/http_server_html_template/src/build.rs [hunk 5]
```diff
@@ -0,0 +1,3 @@
+pub fn main() {
+    let _ = pluto_templating::initialize_templating_engine("views");
```
Comment (pu0238) [inline examples/src/http_server_html_template/src/build.rs:2]: we shouldnt use _ in here try to handdle error in someway or unwrap it if it realy should drop one

File: examples/src/http_server_html_template/src/bootstrap.rs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+#[update]
+async fn http_request_update(req: RawHttpRequest) -> RawHttpResponse {
+    bootstrap(http_serve!(), req).await
+}
+
+async fn bootstrap(mut app: HttpServe, req: RawHttpRequest) -> RawHttpResponse {
+    let router = ROUTER.with(|r| r.borrow().clone());
+    let cors = Cors::new()
```
Comment (pu0238) [inline examples/src/http_server_html_template/src/bootstrap.rs:36]: why our feontend is using cors? is this needed in here?

**Review by pu0238 — CHANGES_REQUESTED** [review]

File: src/pluto_templating/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+use std::path::PathBuf;
+
+/// This is a method that initializes the templating engine in a Pluto project.
+///
+/// Typically it is meant to be used in a build script (`build.rs` file).
+pub fn initialize_templating_engine(templates_dir: &str) -> Result<(), ructe::RucteError> {
+    let out_dir = format!("{}", std::env::var("OUT_DIR").expect("No source path set."));
+    let mut engine = ructe::Ructe::new(PathBuf::from(out_dir)).unwrap();
+    match engine.compile_templates(templates_dir) {
+        Err(_) => return Ok(()),
```
[the remark this answers is not in the document]
Comment (Reyth3) [reply src/pluto_templating/src/lib.rs:10]: In this build script that we run on `cargo build` do we want to throw errors when there are no template files in the project yet?

File: examples/src/http_server_html_template/views/index.rs.html (hunk 4 above)
Comment (Reyth3) [reply examples/src/http_server_html_template/views/index.rs.html:2]: Yes, no parameters from rust to html in this simple example

File: src/pluto_templating/src/lib.rs (hunk 7 above)
[the remark this answers is not in the document]
Comment (pu0238) [reply src/pluto_templating/src/lib.rs:10]: Sure i bealive user should know that he do not created templates yet

File: src/pluto/src/view.rs (hunk 3 above)
Comment (pu0238) [inline src/pluto/src/view.rs:13]:
```suggestion
        let mut headers = HashMap::from([
            ("Content-Type".to_string(), "text/html".to_string()),
        ]);
```

File: src/pluto/src/http.rs [hunk 8]
```diff
@@ -388,4 +388,4 @@ impl HttpServe {
             }
         }
     }
-}
\ No newline at end of file
+}
```
Comment (pu0238) [inline src/pluto/src/http.rs:391]: let's leave empty line at end of the file
```suggestion
}

```

**Review by pu0238 — CHANGES_REQUESTED** [review]

**Review by pu0238 — APPROVED** [review]

**Pull request merged**